issue_11/script.js

```diff
@@ -0,0 +1,61 @@
+function startLoading() {
+    const loadingImg = document.createElement('img');
+    loadingImg.src = 'loading-circle.gif';
+    loadingImg.alt = 'ローディング画像';
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function stopLoading() {
+    document.getElementById('js-loading').remove();
+}
+
+function createLists(resolvedData) {
+    const ul = document.getElementById('js-ul');
+    const fragment = document.createDocumentFragment();
+    for (const item of resolvedData) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.href = item.to;
+        a.textContent = item.text;
+        img.src = item.img;
+        img.alt = item.alt;
+
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    ul.appendChild(fragment);
+}
+
+
+async function callJson() {
+    const url = 'https://api.json-generator.com/templates/NluaaELSLhVe/data?access_token=ambjpfxjl00e50wa639kwndq1ofq3iuykdrv98ge';
+    const response = await fetch(url);
+    const result = await response.json();
+    return result.data;
+}
+
+function getJsonData() {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve(callJson());
+        }, 3000);
+    });
+}
+
+async function tryGetJsonData() {
+    startLoading();
+    try {
+        return await getJsonData();
+    } catch (error) {
+        console.error(error);
+    } finally {
+        stopLoading();
+    }
+}
```

Comment:
IMO
I share what I learned from school student:)
> - When the data is returned, evaluate if the data content is in there. If not, say "data is empty".

```suggestion

async function callJson() {
  try {
    const url =
      "https://api.json-generator.com/templates/NluaaELSLhVe/data?access_token=ambjpfxjl00e50wa639kwndq1ofq3iuykdrv98ge";
    //const url = "https://mocki.io/v1/52fce1d6-6ba9-4f97-a074-3b27b00ab8e0"; //空配列を用意してみました
    const response = await fetch(url);
    const result = await response.json();
    return result.data;
  } catch (e) {
    throw new Error(e);
  }
}

async function tryGetJsonData() {
  startLoading();

  try {
    const data = await callJson();
    const ul = document.getElementById("js-ul");
    
    if (data.length === 0) {
      ul.textContent = "データが空です";
    }
    return data;
  } catch (e) {
    createErrorMessage(e); //別で関数を用意するとわかりやすいです
  } finally {
    stopLoading();
  }
}
```

仕様にはないので修正はお任せいたします🙇‍♀️

Reply:
The function name is also
- callJson() → fetchData()
- tryGetJsonData() →fetchListData()

My teacher taught me that it is suitable.
What do you think?

Reply:
Thank you for your suggestion.
I have looked into error handling, and in this case, I think there are two possible errors: one is an error in fetching, and the other is an error in the content of the retrieved data.
Therefore, I wrote error handling for each of them.
https://github.com/yoshikawa-akr/js-lessons/pull/11/commits/da093d735a937d6a339d9d0d7bdbbe425570f1cc

---
I think the function name you suggested fits better!
We have corrected it.:sparkles:
https://github.com/yoshikawa-akr/js-lessons/pull/11/commits/6d2188955df1fa345addddb9b4c9212059245f12

Reply:
>  I think there are two possible errors: one is an error in fetching, and the other is an error in the content of the retrieved data.

I agree with you.
I think the part about displaying it on the screen is OK!


---------------------
but I have two comments this parts.
```
if (listData && listData.length === 0) {
    throw new Error('データがありません');
 }
```

 ### 1.  No error occurs when data is empty

I also learned about it from the review.

- Quoting the teacher's comments in JS Assignment 15 (on Discord)
> Empty data should not be an error. It is just an initial state where no data is registered. For example, in the case of a Twitter account, it is easy to understand that the account has never been tweeted about.

So, I think we should just display the error on the screen.　


### 2. Q. Is `if (listData.length === 0)` not good enough?
If you need me to, I would love to know why.

Reply:
Thanks for your comments:sparkles:
### 1. https://github.com/yoshikawa-akr/js-lessons/pull/11/commits/c67fa0a94bc6bbd3a2097578f8425a8ff7f13b82
I see, that was a good learning experience.
Fixed it to simply display the message without throwing the error.

### 2. https://github.com/yoshikawa-akr/js-lessons/pull/11/commits/2fdac409dbd8f3b0b5b92861dd6b42d062f9ae14
The error message did not appear as I expected it to, so I had to describe it.
However, I was able to solve the problem by removing `createErrorMessage`.
It was because I had passed `listDataError` as an argument to `createErrorMessage`, so there was no need to include `listData` as a condition in the if statement.

Reply:
修正ありがとうございます！CodeSandboxの更新も、毎回助かります🙇‍♀️
（日本語で失礼します🙇‍♀️）

データが空の時にコンソールにエラーを表示させたいときは、以下のようにすべきだと考えます。
```
    if (listData.length === 0) {
      displayMessage("データがありません");
      console.error("データがありません"); //追加
    }
```

なぜなら修正いただいたコードだと、

①サーバーエラーの時のコンソールには、throw new Errorで投げたエラーと、リストが取得できないエラーが表示
![Screenshot at Mar 18 19-53-45](https://user-images.githubusercontent.com/67426438/158991410-50eee969-c506-4564-852c-392ab79f04ec.png)


②空の配列を取得した時のコンソールは、throw new Errorでエラーを投げない実装に変えたため、エラーがコンソールに表示されない
![Screenshot at Mar 18 19-56-20](https://user-images.githubusercontent.com/67426438/158991423-69c170e1-30dd-4c2a-8545-a736b712dd14.png)

からです。


もし、私の認識が間違えていましたらご指摘いただけると嬉しいです。
ご確認よろしくお願いいたします！

Reply:
Thank you for explaining it so clearly.:sparkles:
I was wondering if I didn't have to describe the data being empty, since you mentioned earlier that it is not an error.
However, after you pointed this out, I felt it would be easier to understand if the data was output to the console, so I have modified it. 
Thank you. 👍 
https://github.com/yoshikawa-akr/js-lessons/pull/11/commits/3946aa06a126afe91559537ddacf7f14058c58d7

Reply:
修正ありがとうございます！
すみません、いただいたコメントを読んで、console.logの方が適していると思いました。これだとエラー出してしまってますね💦

Reply:
Thank you for your repeated comments.
I should have realized this myself when replying.
Sorry for the trouble. 
I have corrected it.
https://github.com/yoshikawa-akr/js-lessons/pull/11/commits/ef7bcc984345045c4f65c6702fab7e42ee279b43